### PR TITLE
Split Condition/CondTree out of GenericPackageDescription.

### PR DESCRIPTION
**Pull request by ezyang**
Signed-off-by: Edward Z. Yang <ezyang@cs.stanford.edu>

### CODE REVIEW

**Comment by hvr** [comment]
Btw, since you're looking at this part of the code: any opinion on #2735?

File: Cabal/Distribution/Types/CondTree.hs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+module Distribution.Types.CondTree (
+    CondTree(..),
+) where
+
+import Prelude ()
+import Distribution.Compat.Prelude
+
+import Distribution.Types.Condition
+
+data CondTree v c a = CondNode
```
Comment (23Skidoo) [inline Cabal/Distribution/Types/CondTree.hs]: Can you please add a Haddock comment here while you're at it?

File: cabal-install/Distribution/Client/JobControl.hs [hunk 2]
```diff
@@ -1,3 +1,4 @@
+--
```
Comment (23Skidoo) [inline cabal-install/Distribution/Client/JobControl.hs]: Committed by mistake?

File: Cabal/Distribution/Types/CondTree.hs [hunk 3]
```diff
@@ -16,10 +19,30 @@ import Distribution.Types.Condition
...
-    , condTreeComponents  :: [( Condition v
-                              , CondTree v c a
-                              , Maybe (CondTree v c a))]
+    , condTreeComponents  :: [CondBranch v c a]
     }
     deriving (Show, Eq, Typeable, Data, Generic, Functor, Foldable, Traversable)
 
 instance (Binary v, Binary c, Binary a) => Binary (CondTree v c a)
+
+data CondBranch v c a = CondBranch
```
Comment (23Skidoo) [inline Cabal/Distribution/Types/CondTree.hs]: Would be nice with a Haddock comment here as well.

**Comment by 23Skidoo** [comment]
LGTM modulo minor comments.

File: cabal-install/Distribution/Client/JobControl.hs (hunk 2 above)
Comment (ezyang) [inline cabal-install/Distribution/Client/JobControl.hs]: Yep.

**Pull request merged**